pkg/tsdb/cloudwatch/time_series_query.go

```diff
@@ -4,126 +4,122 @@ import (
 	"context"
 	"fmt"
 
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana-plugin-sdk-go/data"
 	"github.com/grafana/grafana/pkg/infra/log"
-	"github.com/grafana/grafana/pkg/tsdb"
-	"github.com/grafana/grafana/pkg/util/errutil"
 	"golang.org/x/sync/errgroup"
 )
 
-func (e *cloudWatchExecutor) executeTimeSeriesQuery(ctx context.Context, queryContext *tsdb.TsdbQuery) (*tsdb.Response, error) {
+func (e *cloudWatchExecutor) executeTimeSeriesQuery(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
 	plog.Debug("Executing time series query")
-	startTime, err := queryContext.TimeRange.ParseFrom()
-	if err != nil {
-		return nil, errutil.Wrap("failed to parse start time", err)
-	}
-	endTime, err := queryContext.TimeRange.ParseTo()
-	if err != nil {
-		return nil, errutil.Wrap("failed to parse end time", err)
-	}
-	if !startTime.Before(endTime) {
-		return nil, fmt.Errorf("invalid time range: start time must be before end time")
-	}
 
-	requestQueriesByRegion, err := e.parseQueries(queryContext, startTime, endTime)
-	if err != nil {
-		return nil, err
-	}
+	resp := backend.NewQueryDataResponse()
 
-	if len(requestQueriesByRegion) == 0 {
-		return &tsdb.Response{
-			Results: make(map[string]*tsdb.QueryResult),
-		}, nil
-	}
+	for _, q := range req.Queries {
+		startTime := q.TimeRange.From
+		endTime := q.TimeRange.To
+		if !startTime.Before(endTime) {
+			return nil, fmt.Errorf("invalid time range: start time must be before end time")
```

Comment:
Maybe this is something that could be taken care of by the SDK? Is it even possible to achieve this using the timepicker? Maybe only by hacking the dashboard.json